web/src/click_handlers.js

```diff
@@ -386,10 +387,18 @@ export function initialize() {
     // RESOLVED TOPICS
     $("body").on("click", ".message_header .on_hover_topic_resolve", (e) => {
         e.stopPropagation();
+        const element = e.target;
+        const topic_name = $(e.target).attr("data-topic-name");
+        // Update class to show the spinner
+        element.classList.remove("fa-check");
+        element.classList.add("fa-spinner");
         const $recipient_row = $(e.target).closest(".recipient_row");
         const message_id = rows.id_for_recipient_row($recipient_row);
-        const topic_name = $(e.target).attr("data-topic-name");
         message_edit.toggle_resolve_topic(message_id, topic_name);
+        if (resolved_topic.is_resolved(topic_name)) {
```

Comment:
This if condition provides a double verification, as using `message_edit.toggle_resolve_topic(message_id, topic_name);` changes the class from `.on_hover_topic_resolve` to `.on_hover_topic_unresolve`. This change automatically removes the spinner.